components/skus/browser/rs/lib/src/sdk/credentials/present.rs

```diff
@@ -61,6 +61,50 @@ where
                 let name = "sku#".to_string() + &item.sku;
 
                 let (expires_at, presentation) = match item.credential_type {
+                    CredentialType::TimeLimitedV2 => {
+                        let tlv2_cred = self
+                            .matching_time_limited_v2_credential(&item.id) // get tokens for today
+                            .await?
+                            .ok_or(InternalError::ItemCredentialsMissing)?;
+                        let unblinded_creds =
+                            tlv2_cred.unblinded_creds.ok_or(InternalError::ItemCredentialsMissing)?;
+
+                        // retrieve the next unspent token
+                        let (i, cred) = unblinded_creds
+                            .into_iter()
+                            .enumerate()
+                            .find(|(_i, cred)| !cred.spent)
+                            .ok_or(InternalError::OutOfCredentials)?;
+
+                        let issuer = self.encode_issuer_id(&order.merchant_id, &item.sku)?;
+
+                        let verification_key =
+                            cred.unblinded_cred.derive_verification_key::<Sha512>();
+                        // FIXME change the payload we're creating the binding with
+                        let signature =
+                            verification_key.sign::<HmacSha512>(issuer.as_bytes()).encode_base64();
+
+                        let redemption = json!({
+                            "validFrom": tlv2_cred.valid_from.to_string(),
+                            "validTo": tlv2_cred.valid_to.to_string(),
+                            "issuer": issuer,
+                            "t": cred.unblinded_cred.t,
+                            "signature": signature,
+                        });
+
+                        // burn that cred
+                        self.client.spend_time_limited_v2_item_cred(&item.id, i).await?;
+
+                        // our presentation/redemption
+                        (
+                            Some(format!(
+                                "{}",
+                                tlv2_cred.valid_to.format("%a, %d %b %Y %H:%M:%S GMT")
```

Comment:
why does valid to get duplicated here?

Reply:
This is the same wrapping mechanism that tlv1 credentials use, which tacks the valid_to onto the credential so that the merchant can quickly determine without introspecting the presentation if this is expired or not.

Reply:
hm I'm generally not a fan of duplicating info in schemas since then you need to account for cases where the two are not the same. on the server side is the validFrom / validTo checked against the current time?